utilities/io_tools/output_tools.py

```diff
@@ -134,7 +134,9 @@ def copy_to_eos(outpath, outfolder=None):
 def write_theory_corr_hist(output_name, process, output_dict, args=None, file_meta_data=None): 
     outname = output_name
     if args is not None and args.postfix is not None:
-        outname += f"_{args.postfix}"
+        # Don't add if the generator name is already modified
+        if args.postfix not in args.generator:
+            outname += f"{args.postfix}"
```

Comment:
--postfix is actually used to add any string to the output, it should not be linked to what args.generator has